rust/arrow/src/array/array.rs

```diff
@@ -195,7 +196,8 @@ pub trait Array: fmt::Debug + Send + Sync + ArrayEqual + JsonEqual {
     /// assert_eq!(array.is_valid(1), false);
     /// ```
     fn is_valid(&self, index: usize) -> bool {
-        self.data().is_valid(self.data().offset() + index)
+        let data = self.data_ref();
```

Comment:
Noticed these while benchmarking, avoiding the clone in `data()` doubled the performance when sorting a bigger dataset by multiple columns. This should also improve several other kernels.